src/v/debug_bundle/debug_bundle_service.cc

```diff
@@ -329,13 +329,36 @@ service::rpk_debug_bundle_path(job_id_t job_id) {
     co_return _rpk_process->_output_file_path;
 }
 
-ss::future<result<void>> service::delete_rpk_debug_bundle() {
+ss::future<result<void>> service::delete_rpk_debug_bundle(job_id_t job_id) {
     if (ss::this_shard_id() != service_shard) {
-        co_return co_await container().invoke_on(service_shard, [](service& s) {
-            return s.delete_rpk_debug_bundle();
-        });
+        co_return co_await container().invoke_on(
+          service_shard,
+          [job_id](service& s) { return s.delete_rpk_debug_bundle(job_id); });
+    }
+    auto status = process_status();
+    if (!status.has_value()) {
+        co_return error_info(error_code::debug_bundle_process_never_started);
+    }
+    if (status == debug_bundle_status::running) {
+        co_return error_info(error_code::debug_bundle_process_running);
```

Comment:
Are we not concerned about `debug_bundle_status::error`? I guess the file could still exist in case of a en error.

I prefer a covering switch to document the explicit decision.